### PR TITLE
Adding dev server config

### PR DESCRIPTION
**Pull request by jdltechworks**
@kutenai 

Added codes to make the devserver to use the dev configuration

### CODE REVIEW

**Comment by kutenai** [comment]
Please use paths

File: generators/app/templates/config/dev.js [hunk 1]
```diff
@@ -5,61 +5,52 @@ import merge from 'webpack-merge'
 
 
 import baseConfig, { paths } from './base.js'
-paths.build = path.resolve('../dist.dev');
```
Comment (kutenai) [inline generators/app/templates/config/dev.js:8]: I keep adding this and then you go in and remove it... stop doing that

File: generators/app/templates/config/dev.js [hunk 2]
```diff
@@ -5,61 +5,52 @@ import merge from 'webpack-merge'
...
             devtool: 'source-map',
             entry: {
-                'app': './src/index.js'
+                'app': [
+                    join(__dirname, '../dev/index.js'),
+                ]
             },
             output: {
-                path: paths.build,
+                path: resolve('dist.dev/js'),
```
Comment (kutenai) [inline generators/app/templates/config/dev.js]: I don't want to see hard-code paths!

File: generators/app/templates/config/dev.js [hunk 3]
```diff
@@ -5,61 +5,52 @@ import merge from 'webpack-merge'
...
                 filename: '[name].js',
                 libraryTarget: 'var',
             },
             plugins: [
                 // Creates a manifest file that is Django compatible and read by the Django modules
                 new BundleTracker({
                     //path: path.resolve('lib'),
-                    path: 'lib',
+                    path: resolve('dist.dev'),
```
Comment (kutenai) [inline generators/app/templates/config/dev.js]: Hard coded path

File: generators/app/templates/config/dev.js [hunk 4]
```diff
@@ -5,61 +5,52 @@ import merge from 'webpack-merge'
...
-
     return merge.strategy(strategy)(
-        baseConfig(env),
+        baseConfig(),
         {
             devtool: 'source-map',
             entry: {
-                'app': './src/index.js'
+                'app': [
+                    join(__dirname, '../dev/index.js'),
```
Comment (kutenai) [inline generators/app/templates/config/dev.js]: Hard coded path..
join(path.dev, 'index.js') would be better

File: generators/app/templates/config/devserver.js [hunk 5]
```diff
@@ -1,97 +1,48 @@
...
+                'app': [
+                    //webpack 2 reference
+                    //https://webpack.js.org/guides/hmr-react/
+                    //React hmr entry
+                    'react-hot-loader/patch',
+                    //Dev server bundle
+                    `webpack-dev-server/client?${PROTOCOL}://${DOMAIN}:${PORT}`,
+                    //Only reload successful updates
+                    'webpack/hot/only-dev-server',
+                    join(__dirname, '../dev/index.js')
```
Comment (kutenai) [inline generators/app/templates/config/devserver.js]: join(paths.dev, 'index.js')

File: generators/app/templates/config/devserver.js [hunk 6]
```diff
@@ -1,97 +1,48 @@
...
-        partials.htmlWebpack(paths),
-        partials.copyAssets(paths),
-        {
-            plugins: [
-                new webpack.DefinePlugin({
-                    DEV_LOGGING: JSON.stringify(true),
-                }),
-            ],
-        }
+        partials.htmlWebpack('./dev/templates')
```
Comment (kutenai) [inline generators/app/templates/config/devserver.js]: join(paths.dev, 'templates')

File: generators/app/templates/config/partials.js [hunk 7]
```diff
@@ -69,25 +69,22 @@ export const lintJavaScript = ({ include, exclude, options }) => ({
...
+});
 
 // PLUGINS
 export const htmlWebpack = dir => ({
     plugins: [
         new HtmlWebpackPlugin({
             title: 'Welcome to <%= appname %>',
             inject: 'body',
-            filename: path.join(paths.build, 'index.html'),
             template: path.join(__dirname, '../dev/templates/index.ejs'),
```
Comment (kutenai) [inline generators/app/templates/config/partials.js]: join(paths.dev, 'templates/index.ejs')

File: generators/app/templates/webpack.config.babel.js [hunk 8]
```diff
@@ -1,4 +1,13 @@
-export default env => {
-    return require(`./config/${env}.js`)({env: env})
-}
+export default (env) => {
+    // Let's check if the --env has a serve string
+    // this is one way we can activate devserve
+    const serve = env.includes('serve')
```
Comment (kutenai) [inline generators/app/templates/webpack.config.babel.js]: Why do this? 
I already had a mechanism where we can merge and override previous configs.. so, we can have a devserver.js that merges in additional stuff on top of dev.src, we don't need this extra stuff here.

this is inconsistent with the rest of the logic. The logic is "dev" specifies the filename, I don't want a "special case" for server.. special cases are not special, they are bad.

**Review by kutenai — CHANGES_REQUESTED** [review]

**Comment by jdltechworks** [comment]
@kutenai 

Removed hardcoded paths and special case in the webpack.config.babel.js

included options to use the devserver config

**Pull request merged**